src/Features/Core/Portable/Completion/Providers/AbstractContextVariableArgumentProvider.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    internal abstract class AbstractContextVariableArgumentProvider : ArgumentProvider
+    {
+        public override async Task ProvideArgumentAsync(ArgumentContext context)
+        {
+            if (context.PreviousValue is not null)
+            {
+                return;
+            }
+
+            var semanticModel = await context.Document.GetRequiredSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var symbols = semanticModel.LookupSymbols(context.Position, name: context.Parameter.Name);
+            foreach (var symbol in symbols)
+            {
+                // Currently we check for an exact type match before using a variable from context. As we hone the
+                // default argument provider heuristics, we may alter the definition of "in scope" as well as the type
+                // and name check(s) that occur.
+                if (SymbolEqualityComparer.Default.Equals(context.Parameter.Type, symbol.GetSymbolType()))
```

Comment:
This argument provider is not part of this pull request. I added documentation to it since it came up earlier, but behavior changes would be out of scope.

Reply:
this behavior is fine with me.  i like the comment that makes it clear that this ist just a reasonable starting point, and we can certainly improve just this logic orthogonally to everything else.